ref/application-manifest.md

```diff
@@ -162,3 +164,33 @@ The manifest and extra data in our example are placed next to the application pa
 │       └── data.bin
 └── manifest.yaml
 ```
+
+## Hooks
+
+Hooks allow you to run custom scripts when a certain event is triggered in a container life cycle. Please refer to [Hooks](TBD) for more details about the usage of hooks in an application.
+
+## Bootstrap
+
+An application bootstrap can be fine tuned through the `bootstrap` field.
+
+The `bootstrap` attribute includes the following field definitions:
+
+Name                  | Value type | Description
+----------------------|------------|-------------------------
+`keep`                |  array     | Contents under the [APP_DIR](TDB) directory to be preserved in the application image after the bootstrap is finished. Wildcard pattern is supported, please refer to the [pattern syntax](https://golang.org/pkg/path/filepath/#Match) for more details.
+
+Except for the metadata `manifest` and `hooks` directory (if present), all contents under the `APP_DIR` directory are removed by default after the application bootstrap is finished for the optimum application size. If files under the `APP_DIR` directory are required by a certain hook for the regular container runtime, you need to include them in the application image. For example:
```

Comment:
```suggestion
To minimise the application size, most contents under the `APP_DIR` directory are removed when the application bootstrap is finished. By default, only the `manifest` directory that contains the metadata and the `hooks` directory (if present) are preserved. If a hook requires any other files under the `APP_DIR` directory during the regular container runtime, you must include them in the application image. 

For example:
```

Reply:
I think I should have made it clear in my 1st iteration at this point to avoid confusion.
There is no `manifest` directory at all we see as metadata in the application folder. 

Typically, an application folder contains:
- manifest.yaml
- app.apk
- hooks (if presents)
- other stuff that is required by hooks (if presents)
- extra-data (if presents)
 
Only the "manifest.yaml" file and "hooks" directory are metadata to an application to be created in AMS (yes, we see the `hooks` directory as the metadata). For all other stuff, users have to make a decision to keep them in the app image or not. 

How about:

. By default, only the `manifest` directory that contains the metadata and the `hooks` directory (if present) are preserved`
->
. By default, only the metadata are preserved, they are:
- manifest.yaml
- hooks directory (if present)

Reply:
```suggestion
To minimise the application size, most contents under the `APP_DIR` directory are removed when the application bootstrap is finished. By default, only the metadata content is preserved (the `manifest.yaml` file and the `hooks` directory, if present). If a hook requires any other files under the `APP_DIR` directory during the regular container runtime, you must include them in the application image. 

For example:
```

Reply:
Thanks, that makes it much clearer! I was wondering why we suddenly talk about a manifest directory, but didn't look it up. ;)